cluster-autoscaler/cloudprovider/aws/auto_scaling.go

```diff
@@ -94,6 +145,63 @@ func (m *autoScalingWrapper) getAutoscalingGroupsByNames(names []string) ([]*aut
 	return asgs, nil
 }
 
+func (m autoScalingWrapper) populateLaunchConfigurationInstanceTypeCache(autoscalingGroups []*autoscaling.Group) error {
+	var launchConfigToQuery []*string
+
+	m.launchConfigurationInstanceTypeCache.jitterClock.Lock()
+	m.launchConfigurationInstanceTypeCache.jitterClock.jitter = true
+	m.launchConfigurationInstanceTypeCache.jitterClock.Unlock()
+	for _, asg := range autoscalingGroups {
+		if asg == nil {
+			continue
+		}
+		if asg.LaunchConfigurationName == nil {
+			continue
+		}
+		_, found, _ := m.launchConfigurationInstanceTypeCache.GetByKey(*asg.LaunchConfigurationName)
+		if found {
+			continue
+		}
+		launchConfigToQuery = append(launchConfigToQuery, asg.LaunchConfigurationName)
+	}
+	m.launchConfigurationInstanceTypeCache.jitterClock.Lock()
+	m.launchConfigurationInstanceTypeCache.jitterClock.jitter = false
+	m.launchConfigurationInstanceTypeCache.jitterClock.Unlock()
+
+	// List expire old entries
+	_ = m.launchConfigurationInstanceTypeCache.List()
+	if len(launchConfigToQuery) == 0 {
+		klog.V(4).Infof("%d launch configurations already in cache", len(autoscalingGroups))
+		return nil
+	}
+	klog.V(4).Infof("%d launch configurations to query", len(launchConfigToQuery))
+
+	for i := 0; i < len(launchConfigToQuery); i += 50 {
```

Comment:
This looks duplicate as `getInstanceTypeByLCName`. Is it possible to refactor the `getInstanceTypeByLCName ` and make it accept a slice of LC names? Then we can reuse it.